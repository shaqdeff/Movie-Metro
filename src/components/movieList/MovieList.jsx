import React, { useState } from 'react';
import Slider from 'react-slick';
import { useDispatch, useSelector } from 'react-redux';
import {
  getAllMovies,
  getAllShows,
  fetchAsyncMovies,
  fetchAsyncShows,
} from '../../redux/moviesSlice/moviesSlice';
import MovieCard from '../movieCard/MovieCard';
import Settings from '../../settings';
import './movieList.scss';

/* eslint-disable operator-linebreak */
/* eslint-disable react/jsx-props-no-spreading */
const MovieList = () => {
  const movies = useSelector(getAllMovies);
  const shows = useSelector(getAllShows);
  let renderMovies = '';
  let renderShows = '';

  renderMovies =
    movies.Response === 'True' ? (
      movies.Search.map((movie) => (
        <MovieCard key={movie.imdbID} data={movie} />
      ))
    ) : (
      <div className="movies-error">
        <h3>{movies.Error}</h3>
      </div>
    );

  renderShows =
    shows.Response === 'True' ? (
      shows.Search.map((movie) => <MovieCard key={movie.imdbID} data={movie} />)
    ) : (
      <div className="shows-error">
        <h3>{shows.Error}</h3>
      </div>
    );

  const [term, setTerm] = useState('');
  const dispatch = useDispatch();

  const submitHandler = (e) => {
    e.preventDefault();
    dispatch(fetchAsyncMovies(term));
    dispatch(fetchAsyncShows(term));
    setTerm('');
  };

  return (
    <div>
      <div className="search-bar">
        <form onSubmit={submitHandler}>
          <input
            type="text"
            value={term}
            placeholder="search movies"
            onChange={(e) => setTerm(e.target.value)}
          />
          <button type="submit">
            <i className="fa fa-search" />
          </button>
        </form>
      </div>
      <div className="movie-wrapper">
        <div className="movie-list">
          <h2>Movies</h2>
          <div className="movie-container">
            <Slider {...Settings}>{renderMovies}</Slider>
          </div>
        </div>
        <div className="shows-list">
          <h2>Shows</h2>
          <div className="movie-container">
            <Slider {...Settings}>{renderShows}</Slider>
          </div>
        </div>
      </div>
    </div>
  );
};

export default MovieList;
