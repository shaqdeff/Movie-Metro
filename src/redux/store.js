import { configureStore } from '@reduxjs/toolkit';
import moviesReducer from './moviesSlice/moviesSlice';

const store = configureStore({
  reducer: {
    movies: moviesReducer,
  },
});

export default store;
