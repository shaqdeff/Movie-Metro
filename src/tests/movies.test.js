import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MovieList from '../components/movieList/MovieList';

it('renders correctly', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <MovieList />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
