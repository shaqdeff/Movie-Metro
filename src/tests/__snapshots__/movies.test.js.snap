// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="search-bar"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="search movies"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>
  <div
    className="movie-wrapper"
  >
    <div
      className="movie-list"
    >
      <h2>
        Movies
      </h2>
      <div
        className="movie-container"
      >
        <div
          className="slick-slider slick-initialized"
        >
          <div
            className="slick-list"
          >
            <div
              className="slick-track"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                  "WebkitTransition": "",
                  "msTransform": "translateX(0px)",
                  "opacity": 1,
                  "transform": "translate3d(0px, 0px, 0px)",
                  "transition": "",
                }
              }
            >
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-current"
                data-index={0}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="movies-error"
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <h3 />
                  </div>
                </div>
              </div>
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-cloned"
                data-index={1}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="movies-error"
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <h3 />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="shows-list"
    >
      <h2>
        Shows
      </h2>
      <div
        className="movie-container"
      >
        <div
          className="slick-slider slick-initialized"
        >
          <div
            className="slick-list"
          >
            <div
              className="slick-track"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                  "WebkitTransition": "",
                  "msTransform": "translateX(0px)",
                  "opacity": 1,
                  "transform": "translate3d(0px, 0px, 0px)",
                  "transition": "",
                }
              }
            >
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-current"
                data-index={0}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="shows-error"
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <h3 />
                  </div>
                </div>
              </div>
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-cloned"
                data-index={1}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="shows-error"
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <h3 />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
